src/cluster/src/server.rs

```diff
@@ -168,17 +169,54 @@ where
         )
         .await?;
 
-        let idle_merge_effort = isize::cast_from(config.idle_arrangement_merge_effort);
+        let mut worker_config = WorkerConfig::default();
+
+        let idle_merge_effort = config.idle_arrangement_merge_effort;
         // We want a value of `0` to disable idle merging. DD will only disable idle merging if we
         // configure the effort to `None`, not when we set it to `Some(0)`.
         let idle_merge_effort = (idle_merge_effort > 0).then_some(idle_merge_effort);
 
-        let mut worker_config = WorkerConfig::default();
+        // By default, se only set the idle_merge_effort
         differential_dataflow::configure(
             &mut worker_config,
-            &differential_dataflow::Config { idle_merge_effort },
+            &differential_dataflow::Config {
+                idle_merge_effort: idle_merge_effort.map(CastFrom::cast_from),
+            },
         );
 
+        // We set a custom exertion logic for idle_merge_effort of 0, and proportionality > 0.
+        // This avoids turning on proportionality for replicas with default idle merge effort.
+        if idle_merge_effort.is_none() && config.arrangement_exert_proportionality > 0 {
+            let idle_merge_effort = Some(1000);
+            let arc: ExertionLogic = Arc::new(move |layers| {
```

Comment:
The meaning of three values in `layers` is undocumented, including in differential. It's just a tuple three `usize`s. Can we fix this (either in Materialize or Differential) ? Otherwise the code is much harder to understand.

Reply:
Added some documentation, but I agree that we should make sure it's documented within Differential, too.